sdk/typescript/src/utils/gas-estimation.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { UnserializedSignableTransaction } from '../signers/txn-data-serializers/txn-data-serializer';
+import { Coin, SuiMoveObject } from '../types';
+
+export type TxKind = UnserializedSignableTransaction['kind'];
+type BudgetMap = typeof DEFAULT_GAS_BUDGET_PER_TX_TYPE;
+export type GasBudgetGuessParams<txKind extends TxKind> =
+  txKind extends keyof BudgetMap
+    ? BudgetMap[txKind] extends (...args: any[]) => any
+      ? Parameters<BudgetMap[txKind]> extends [boolean, ...infer Params]
+        ? Params
+        : []
+      : []
+    : [];
+
+export const MIN_GAS_BUDGET = 10;
+export const MAX_GAS_BUDGET = 1_000_000;
```

Comment:
Can we document both of these magic numbers?

Reply:
I figured them out by running txs with too small and too big gas budgets. What should we document? Their meaning? or how we calculated them?

(They don't really affect the calculations for budget only for some edge cases)

Reply:
done